electron/generators/diory-generator.js

```diff
@@ -79,11 +92,51 @@ function getAverageDate(linkedDiorys) {
   )
 }
 
-exports.generateFolderDiory = function generateFolderDiory(folderPath, linkedDiorys = []) {
-  return generateDiory({
-    ...getFirstImage(linkedDiorys),
-    ...getAverageLocation(linkedDiorys),
-    ...getAverageDate(linkedDiorys),
-    ...readFolder(folderPath),
-  })
+/**
+ * What is this? What does it do? Not really self-explaining...
+ * Does this replace link-generator.js generateLink?
+ */
+function generateLinks(dioryLinks) {
+  return Object.entries(dioryLinks).reduce(
+    (obj, [linkKey, { id }]) => ({
+      ...obj,
+      [linkKey]: { id },
+    }),
+    {}
+  )
+}
+
+/**
+ * Generates diograph from folder
+ * @param {string} filePath - Path to file which diory should be generated
+ * @param {object} dioryLinks - Object of diory links
+ *    {
+ *      'filename1': { [diory1 object] },
+ *      'filename2': { [diory2 object] }
+ *    }
+ * @return {object} - Folder diograph generated from file
+ *    {
+ *      id: 'some-id',
+ *      text: 'text',
+ *      image: 'image'
+ *      ...
+ *      modified: 'modified'
+ *      links: [
+ *        'filename1': { [diory object] },
+ *        'filename2': { [diory2 object] }
+ *      ]
+ *    }
+ */
+exports.generateDioryFromFolder = function (folderPath, dioryLinks = {}) {
+  const linkedDiorys = Object.values(dioryLinks)
+  return {
+    ...generateDiory({
+      ...getFirstImage(linkedDiorys),
+      ...getAverageLocation(linkedDiorys),
+      ...getAverageDate(linkedDiorys),
+      ...readFolderMetadata(folderPath),
+      ...generateLinks(dioryLinks),
+    }),
+    ...(isEmpty(dioryLinks) ? {} : { links: dioryLinks }),
```

Comment:
diory do have dioryLinks, but links; and this overwrites ...generateLinks(dioryLinks) that generates links.

Reply:
thus dioryLinks should be converted to links in diograph generator.